File: burstLibrary/src/main/java/com/harry1453/burst/explorer/service/BurstServiceProvider.java
package com.harry1453.burst.explorer.service;

import com.harry1453.burst.explorer.repository.ConfigRepository;

public final class BurstServiceProvider {

    private final ObjectService objectService;
    private final ConfigRepository configRepository;

    BurstServiceProvider(ObjectService objectService, ConfigRepository configRepository) {
        this.objectService = objectService;
        this.configRepository = configRepository;
    }

    public BurstBlockchainService getBurstBlockchainService() {
        return new NodeBlockchainService(configRepository, objectService);
    }

    public BurstInfoService getBurstInfoService() {
        return new RepoInfoService(objectService);
    }

    public BurstNetworkService getBurstNetworkService() {
        return new PoCCNetworkService(objectService);
    }

    public BurstPriceService getBurstPriceService() {
        return new CMCPriceService(objectService);
    }
}

File: burstLibrary/src/main/java/com/harry1453/burst/explorer/service/BurstServiceProvider.kt
package com.harry1453.burst.explorer.service

import com.harry1453.burst.explorer.repository.ConfigRepository

class BurstServiceProvider internal constructor(private val objectService: ObjectService, private val configRepository: ConfigRepository) {
    val burstBlockchainService: BurstBlockchainService
        get() = NodeBlockchainService(configRepository, objectService)

    val burstInfoService: BurstInfoService
        get() = RepoInfoService(objectService)

    val burstNetworkService: BurstNetworkService
        get() = PoCCNetworkService(objectService)

    val burstPriceService: BurstPriceService
        get() = CMCPriceService(objectService)
}
